### PR TITLE
SAK-46881 Dropbox user input is lost when validation fails

### PR DESCRIPTION
**Pull request by bbailla2**
In the Drop Box / Resource tools, there are many forms whose fields are all cleared when users submit invalid data (use input loss). When this happens, it can cause a very bad impression of the system.

Preserve user input.

### CODE REVIEW

**Comment by bbailla2** [comment]
When reviewing this PR, it may be helpful to understand the following:

I introduced UserInputPreserver.java, which takes user input from ParameterParser, stores it in State attributes, and in the case of a validation error, puts it back into the Context. UserInputPreserver is abstract, and any views that require validation can add their own implementation.

DropboxMultipleFoldersUploadInputPreserver.java is such an implementation for the "Upload files to multiple Drop Box folders" tab.

CopyrightDelegate.java's variable names were misleading, so I reduced some technical debt

File: content/content-tool/tool/src/webapp/vm/resources/sakai_create_folders.vm [hunk 1]
```diff
@@ -61,14 +63,35 @@
...
 #parse("/vm/resources/sakai_properties_scripts.vm")
+
+<script type="text/javascript">
+	function validateFolderNames()
+	{
+		var textInputs = document.querySelectorAll("#fileInputDiv input[type=text]");
+		var allFolderNamesPresent = true;
+		for (var i=0; i<textInputs.length; i++)
+		{
+			if (textInputs[i].id.indexOf('content${DOT}')==0)
```
Comment (plukasew) [inline content/content-tool/tool/src/webapp/vm/resources/sakai_create_folders.vm:85]: prefer === in javascript

File: content/content-tool/tool/src/webapp/vm/resources/sakai_dropbox_multiple_folders_upload.vm [hunk 2]
```diff
@@ -159,10 +179,54 @@
...
 	{
 		var fileField = document.getElementById('MultipleFolderContent');
 		var nameField = document.getElementById('MultipleFolderDisplayName');
-		nameField.value = fileField.files[0].name;
+
+		## Goal: update the display name field. But if the user provided their own display name, don't overwrite it
+
+#if ($!alreadyUploadedFile)
+		## A file was uploaded to the server, and there was a validation error; track if the file name matches the serverside file
+		var matchesPreviousFile = nameField.value.trim() == "$validator.escapeHtml("$alreadyUploadedFile.getFileName()")".trim();
```
Comment (plukasew) [inline content/content-tool/tool/src/webapp/vm/resources/sakai_dropbox_multiple_folders_upload.vm:194]: Prefer const or let over var

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java [hunk 3]
```diff
@@ -9303,59 +9323,26 @@ public String buildDropboxMultipleFoldersUploadPanelContext(VelocityPortlet port
...
+	        usersDropboxList = inputPreserver.prepareGroupsAndUsersForContext(site_groups, users);
 	    } catch (Exception ex) {
 	        log.error("Exception while getting users collections", ex);
 	    }
 		copyrightChoicesIntoContext(state, context);
 	    context.put("usersDropboxList", usersDropboxList);
+
+	    // Max upload size:
+	    String max_file_size_mb = (String)state.getAttribute(STATE_FILE_UPLOAD_MAX_SIZE);
+	    max_file_size_mb = max_file_size_mb == null ? "20" : max_file_size_mb;
```
Comment (plukasew) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java:9343]: Where does the value 20 come from?

**Review by ern — APPROVED** [review]

**Pull request merged**